### ISSUE TITLE
Transition retry with dynamic segments throws error

### ISSUE THREAD

**Issue by novtor** [emberjs/ember.js#10972]
When retrying a transition with dynamic segments an error is thrown:
_You didn't provide enough string/numeric parameters to satisfy all of the dynamic segments for route_

My scenario: 
1. user goes on /songs?author='Ivan'
2. then /songs/1?author='Ivan'
And then reloads the entire page (with F5)
3. On reloading my ApplicationRoute sees that the user is not logged on so it saves the current transition (_songs/1?author='Ivan'_) and redirects to another route to log the user in. (similar as in http://guides.emberjs.com/v1.10.0/routing/preventing-and-retrying-transitions/)
4. On login success, the transition (_songs/1?author='Ivan'_) is retried and the error is thrown.

If user does only the first step and reloads the page, everything works fine

**Comment by workmanw** [comment]
I think this is related to my issue https://github.com/emberjs/ember.js/issues/10945. I'd be `author` is a `refreshModel` param. If you temporarily remove the `refreshModel` config and retry, does it work?

**Comment by novtor** [comment]
Yes, that is it. Setting the author _queryParams_ as _refreshModel: false_ resolves the issue. Thank you @workmanw

**Comment by novtor** [comment]
Looks like it has been fixed. Perhaps by the #11636. @workmanw please check also your  #10945 if it is fixed?

**Comment by workmanw** [comment]
@novtor I checked in canary and my issues is still not working :( Here is an updated JSBin with Canary links. http://output.jsbin.com/lekegowuqu/1#/post/1/comments?search=awesome

**Comment by novtor** [comment]
You are right @workmanw , the bug persists, I was checking it on a route with refreshModel: false queryParams that's why it worked. 
Looking forwrard on fixing it

**Comment by LexLythius** [comment]
Still not working with Ember 1.13.11.
My workaround for this has been:
````
this.transitionTo(URI_PATH_QUERY_AND_HASH_AS_STRING);
````

**Comment by erichonkanen** [comment]
@LexLythius could you give me a better example of your workaround? I ran into this error when calling `this.refresh()` inside a Route.actions.refresh action... not sure what's going on?!

**Comment by LexLythius** [comment]
@erichonkanen Simply, you have to build the target URL manually -- with all complexities involved. E.g.:
````javascript
this.transitionTo("//yourdomain.com/#/some/path?query1=someValue&query2=another");
````
Useful pieces here are:
 - window.location -- this might be good enough for your workaround;
 - environment variables;
 - controllers' current path, for example `this.container.lookup('controller:application').get('currentPath');`
 - [Route](http://emberjs.com/api/classes/Ember.Route.html)'s  `paramsFor` method and `queryParams` properties;
 - JS native `encodeURIComponent` function.

**Comment by erichonkanen** [comment]
@LexLythius ok awesome ill try that thanks!

**Comment by rwjblue** [comment]
Fix for https://github.com/emberjs/ember.js/issues/10945 was included in 2.4.4.  This should have also been fixed.
